be/src/vec/exec/scan/vfile_scanner.cpp

```diff
@@ -336,12 +336,13 @@ Status VFileScanner::_get_next_reader() {
             for (int i = 0; i < _file_slot_descs.size(); i++) {
                 column_names.push_back(_file_slot_descs[i]->col_name());
             }
-            _cur_reader = new ParquetReader(_profile, _params, range, column_names,
```

Comment:
I think we should not use `_cur_reader` as a pure pointer. Maybe we can resolve this issue by changing this to a unique pointer